### PR TITLE
Inject cache into the  constructor

### PR DESCRIPTION
**Pull request by dkubb**
* This will allow the cache argument to be overridden when necessary and
  removes some tight coupling between Rails.cache and shipping
  calculators.

[https://github.com/MountainRoseHerbs/mrh-spree/issues/1406]

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj this is ready for review.

File: app/models/spree/calculator/shipping/active_shipping/base.rb [hunk 1]
```diff
@@ -153,7 +158,7 @@ def retrieve_timings(origin, destination, packages)
             end
 
             error = Spree::ShippingError.new("#{I18n.t(:shipping_error)}: #{message}")
-            Rails.cache.write @cache_key+"-timings", error #write error to cache to prevent constant re-lookups
+            @cache.write @cache_key+"-timings", error #write error to cache to prevent constant re-lookups
```
Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: Should we use our style here with explicit parens?

File: app/models/spree/calculator/shipping/active_shipping/base.rb [hunk 2]
```diff
@@ -127,7 +132,7 @@ def retrieve_rates(origin, destination, shipment_packages)
             end
 
             error = Spree::ShippingError.new("#{I18n.t(:shipping_error)}: #{message}")
-            Rails.cache.write @cache_key, error #write error to cache to prevent constant re-lookups
+            @cache.write @cache_key, error #write error to cache to prevent constant re-lookups
```
Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: I'd also prefer to use explicit parens here.

File: app/models/spree/calculator/shipping/active_shipping/base.rb (hunk 1 above)
Comment (dkubb) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: I can change more of the line to match our style.

File: app/models/spree/calculator/shipping/active_shipping/base.rb [hunk 3]
```diff
@@ -279,7 +284,7 @@ def cache_key(package)
         end
 
         def retrieve_rates_from_cache package, origin, destination
```
Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: Lets add parens here in the arguments also. Its so adjacent to our change we can easily clean this up a bit.

File: app/models/spree/calculator/shipping/usps/base.rb [hunk 4]
```diff
@@ -52,7 +52,7 @@ def retrieve_rates(origin, destination, shipment_packages)
             end
 
             error = Spree::ShippingError.new("#{I18n.t(:shipping_error)}: #{message}")
-            Rails.cache.write @cache_key, error #write error to cache to prevent constant re-lookups
+            @cache.write @cache_key, error #write error to cache to prevent constant re-lookups
```
Comment (mbj) [inline app/models/spree/calculator/shipping/usps/base.rb]: Lets add parens here also.

**Comment by mbj** [comment]
Code wise I'm fine. I flagged some small style stuff. Will do a validation on PR against mrh-spree.

**Comment by dkubb** [comment]
@mbj I've taken care of all the style issues you flagged.

**Comment by mbj** [comment]
CI was green before you only changed style, AST is identical so I'll merge now.

**Pull request merged**